### PR TITLE
Document tests and clean up runner

### LINKED ISSUE #214: Document test suite
Now that we have a pretty neat test suite I should document how it works and what options it supports.

### PR DESCRIPTION
**Pull request by Nadrieril**
Fixes #214

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Aeneas test suite
```
Comment (sonmarcho) [inline tests/README.md:1]: Small nit: for the title, you should use `##` for the section title (i.e., `## Aeneas test suite`)`, and `###` for the subsections (e.g., `### Running the test suite`).

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# Adding a new test
+
+Adding a new test is easy: just add a `foo.rs` file or a `foo_crate` folder in `tests/src`. `make
+test` will find the new file and call the test runner on it.
+
+To specify options for how your test should be run, see the next section.
+
+Ideally, any non-trivial change to Aeneas should have an accompanying test that exercises the new
+code. The goal of this test suite is to cover a large portion of what Aeneas can do, so we can work
+on it confidently.
```
Comment (sonmarcho) [inline tests/README.md:36]: Maybe give somewhere the command to run a test for a file or a crate? (e.g., `make test-foo.rs` for a file, `make test-bar` for crate `bar`)

File: tests/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+- `[coq,fstar] subdir=misc`: store the output in the `misc` subdirectory for these two backends.
+
+For an up-to-date list of options, see comments in `tests/test_runner/Input.ml`.
+
+# `known-failure` tests
+
+There's a special kind of test that doesn't generate backend code: `//@ known-failure` tests. These
+are meant to record:
+- Cases that produce an error so we don't forget and can fix them later;
+- Cases that _should_ error, to ensure we correctly raise an error and that the error output is nice.
```
Comment (sonmarcho) [inline tests/README.md:60]: Maybe a few words about the `.out` files? (here or above, when you talk about the committed files: I think it's important to mention that they must be committed)

**Review by sonmarcho — APPROVED** [review]
I have a few nitpicks but otherwise looks good!

File: tests/README.md (hunk 1 above)
Comment (Nadrieril) [reply tests/README.md:1]: Why? Is `#` too big?

Comment (Nadrieril) [reply tests/README.md]: Oh, you might have misunderstood: this is not a new section in the main README, this is a new README file

**Review by sonmarcho — APPROVED** [review]

**Pull request merged**